src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -24,6 +24,27 @@ public partial class KubernetesClientConfiguration
         /// </summary>
         public string CurrentContext { get; private set; }
 
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="KubernetesClientConfiguration" /> from config file
+        /// </summary>
+        public static KubernetesClientConfiguration BuildDefaultConfig() {
+            var kubeconfig = Environment.GetEnvironmentVariable("KUBECONFIG");
+            if (kubeconfig != null) {
+                return BuildConfigFromConfigFile(kubeconfigPath: kubeconfig);
+            }
+            if (File.Exists(KubeConfigDefaultLocation)) {
+                return BuildConfigFromConfigFile(kubeconfigPath: KubeConfigDefaultLocation);
+            }
+            var tokenPath = KubernetesClientConfiguration.ServiceaccountPath + KubernetesClientConfiguration.ServiceAccountTokenKeyFileName;
+            if (File.Exists(tokenPath)) {
```

Comment:
Here should be conform with the logic within `InClusterConfig`

Maybe it is better to extract `IsInCluster ` from  `InClusterConfig` and change `InClusterConfig` to `assert IsInCluster`

```
IsInCluster => {
            var host = Environment.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST");
            var port = Environment.GetEnvironmentVariable("KUBERNETES_SERVICE_PORT");

            return (string.IsNullOrWhiteSpace(host) || string.IsNullOrWhiteSpace(port)) ;
}
```

Reply:
done.